internal/rest/client.go

```diff
@@ -191,9 +191,6 @@ func (c *Client) newRequest(ctx context.Context, u *url.URL, body io.Reader) (*h
 		req.Header.Set("Authorization", "Bearer "+c.newAuthToken(u.RawQuery))
 	}
 	req.Header.Set("X-Minio-Time", time.Now().UTC().Format(time.RFC3339))
-	if body != nil {
-		req.Header.Set("Expect", "100-continue")
-	}
```

Comment:
This is actually done to ensure we make sure to not send the body if the server is not able to receive requests @vadmeste 

Do you have evidence on how removing this helps? this has been around for the last 3-4yrs already.

Reply:
> This is actually done to ensure we make sure to not send the body if the server is not able to receive requests @vadmeste

This is internode communication, there is no case when a peer needs to validate an incoming request before allowing to receive the body. I don't think this is needed.

Reply:
> Do you have evidence on how removing this helps? this has been around for the last 3-4yrs already.

Probably better performance for small files with fast NVME disks. For WAN setups, for sure it will have some benefits

Reply:
Can you do some numbers here with some delays and get numbers on why would it matter?